core/src/main/java/com/linecorp/armeria/common/AbstractRequestContextBuilder.java

```diff
@@ -433,6 +438,24 @@ protected final String path() {
         return path;
     }
 
+    /**
+     * Sets the UUID of the request. If not set, it is auto-generated with UUID
+     */
+    public AbstractRequestContextBuilder uuid(UUID uuid) {
+        this.uuid = requireNonNull(uuid, "uuid");
+        return this;
+    }
```

Comment:
Make sure to override this method in `{Client,Service}RequestContextBuilder` and change the return type, e.g.

```java
class ClientRequestContextBuilder ... {
    ...
    @Override
    public ClientRequestContextBuilder uuid(UUID uuid) {
        return (ClientRequestContextBuilder) super.uuid(uuid);
    }
    ...
}
```